packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,614 @@
+      const barHeight: number = Math.max(yBarScale(point.y), 0);
+      if (barHeight < 1) {
+        return <React.Fragment key={point.x}> </React.Fragment>;
+      }
+      return (
+        <rect
+          key={point.x}
+          x={this._isRtl ? xBarScale(point.x) : this.margins.left!}
+          y={yBarScale(point.y) + this._barWidth / 3}
+          width={
+            this._isRtl
+              ? containerWidth - this.margins.right! - Math.max(xBarScale(point.x), 0)
+              : Math.max(xBarScale(point.x), 0) - this.margins.left!
+          }
+          height={this._barWidth}
+          aria-labelledby={`toolTip${this._calloutId}`}
+          aria-label={this._getAriaLabel(point)}
+          role="img"
+          ref={(e: SVGRectElement) => {
+            this._refCallback(e, point.legend!);
+          }}
+          onClick={point.onClick}
+          onMouseOver={this._onBarHover.bind(this, point, colorScale(point.x))}
+          onMouseLeave={this._onBarLeave}
+          onBlur={this._onBarLeave}
+          data-is-focusable={!this.props.hideTooltip}
+          onFocus={this._onBarFocus.bind(this, point, index, colorScale(point.x))}
+          fill={point.color ? point.color : colorScale(point.x)}
+        />
+      );
+    });
+
+    // Removing un wanted tooltip div from DOM, when prop not provided.
+    if (!this.props.showYAxisLablesTooltip) {
+      try {
+        document.getElementById(this._tooltipId) && document.getElementById(this._tooltipId)!.remove();
+        //eslint-disable-next-line no-empty
+      } catch (e) {}
+    }
+    // Used to display tooltip at y axis labels.
+    if (this.props.showYAxisLablesTooltip) {
+      const yAxisElement = d3Select(yElement).call(yBarScale);
+      if (!this.state.tooltipElement) {
+        try {
+          document.getElementById(this._tooltipId) && document.getElementById(this._tooltipId)!.remove();
+          //eslint-disable-next-line no-empty
+        } catch (e) {}
+      }
+      const ytooltipProps = {
+        tooltipCls: this._classNames.tooltip!,
+        id: this._tooltipId,
+        yAxis: yAxisElement,
+      };
+      yAxisElement && this._tooltipOfYAxislabels(ytooltipProps);
+    }
+    return bars;
+  }
+
+  private _closeCallout = () => {
+    this.setState({
+      isCalloutVisible: false,
+    });
+  };
+
+  private _onLegendClick(customMessage: string): void {
+    if (this.state.isLegendSelected) {
+      if (this.state.selectedLegendTitle === customMessage) {
+        this.setState({
+          isLegendSelected: false,
+          selectedLegendTitle: customMessage,
+        });
+      } else {
+        this.setState({
+          selectedLegendTitle: customMessage,
+        });
+      }
+    } else {
+      this.setState({
+        isLegendSelected: true,
+        selectedLegendTitle: customMessage,
+      });
+    }
+  }
+
+  private _onLegendHover(customMessage: string): void {
+    if (this.state.isLegendSelected === false) {
```

Comment:
Kshitij made a change to fix how legend states are maintained. Check in that PR if it is applicable here or not.

Reply:
I will create a separate work item for that.

Reply:
Share the work item id here.

Reply:
https://uifabric.visualstudio.com/iss/_workitems/edit/8225/

Will take all 3 pending review comments in the same work item.